test/client/connection.spec.js

```diff
@@ -111,4 +114,233 @@ describe( 'Connection', () => {
+                <label class="question non-select "><span lang="default" class="question-label active" data-itext-id="/image-default/happy:label">label with image</span><img lang="default" class="active" src="/-/media/get/http/localhost:8989/v1/projects/2/forms/media-spaces/attachments/happy2.png" data-itext-id="/image-default/happy:label" alt="image"><input type="text" name="/image-default/happy" data-type-xml="string"></label>
+                <label class="question non-select "><span lang="default" class="question-label active" data-itext-id="/image-default/unhappy:label">label with image</span><img lang="default" class="active" src="jr://images/un%20happy%202%20v2.png" data-itext-id="/image-default/unhappy:label" alt="image"><input type="text" name="/image-default/unhappy" data-type-xml="string"></label>
+                <fieldset id="or-preload-items" style="display:none;"><label class="calculation non-select "><input type="hidden" name="/image-default/meta/instanceID" data-preload="uid" data-preload-params="" data-type-xml="string"></label></fieldset><fieldset id="or-setvalue-items" style="display:none;"></fieldset>
+            </form>
+        `;
+
+        const model = /* xml */`
+            <model>
+                <instance>
+                    <image-default xmlns:jr="http://openrosa.org/javarosa" xmlns:odk="http://www.opendatakit.org/xforms" xmlns:orx="http://openrosa.org/xforms" id="media-spaces" version="8">
+                        <item>initial</item>
+                        <ann src="/-/media/get/http/localhost:8989/v1/projects/2/forms/media-spaces/attachments/un%20happy%20v2.png">jr://images/un happy v2.png</ann>
+                        <dra src="/-/media/get/http/localhost:8989/v1/projects/2/forms/media-spaces/attachments/indifferent.png">jr://images/indifferent.png</dra>
+                        <happy/>
+                        <unhappy/>
+                        <meta>
+                            <instanceID/>
+                        </meta>
+                    </image-default>
+                </instance>
+                <instance id="last-saved" src="jr://instance/last-saved"/>
+                <instance id="${externalInstanceId}" src="${externalInstanceURL}" />
+            </model>
+        `;
+
+        const hash = 'md5:0aef088ebda239d644130e6bf2255bcf-311fab58efa4fa65318450bf93b70214-7b69544085c4c49cbb77be3890a23995---1';
+
+        const basePath = '-';
+
+        /** @type {Survey} */
+        let transformedSurvey;
+
+        /** @type {string} */
+        let expectedPostBody;
+
+        /** @type {string} */
+        let expectedURL;
+
+        /** @type {string} */
+        let theme;
+
+        beforeEach( async () => {
+            if ( !Object.prototype.hasOwnProperty.call( settings, 'basePath' ) ) {
```

Comment:
Both this and [the other one](https://github.com/enketo/enketo-express/pull/316#discussion_r715807202) you asked about were encountered in the course of writing these tests, and `sinon` won't stub a nonexistent property. I've tried to trace through how `settings` is populated but it's quite a winding path and it felt more efficient to just say "because `sinon` said it wasn't there" 🙁